tensorvis/tensorvisbase/src/datastructures/tensorfield3d.cpp

```diff
@@ -178,6 +178,46 @@ TensorField3D::TensorField3D(const size_t x, const size_t y, const size_t z,
     computeDataMaps();
 }
 
+TensorField3D::TensorField3D(size3_t dimensions, const double *data, const dvec3 &extent,
+                             double sliceCoord)
+    : dimensions_(dimensions)
+    , extent_(extent)
+    , indexMapper_(dimensions)
+    , size_(dimensions.x * dimensions.y * dimensions.z)
+    , rank_(2)
+    , dimensionality_(3)
+    , offset_(dvec3(0.0)) {
+    tensors_.resize(size_);
+    std::memcpy(tensors_.data(), data, size_ * sizeof(double) * 9);
```

Comment:
copy tensors one by one since `glm::dmat3` might have some padding (depending on compile flags). Otherwise you need to add something like `static_assert(sizeof(dmat3) == 9 * sizeof(double), "glm::dmat3 has padding, not supported");`
```c++
tensors_.resize(size_);
for (size_t i = 0; i<size_; ++i) {
    std::copy(data + i * 9, data + (i + 1) * 9, glm::value_ptr(tensors_[i]));
}
```

Reply:
Done.